### PR TITLE
[Debugger] Improve variable visibility test

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
This PR will improve the variable visibility test with the following scenarios:

- [x] The variable count should be zero(0) at the beginning of the main() method.
- [x] All variables should be visible at the last line of the main() method.
- [x] Global variables and their values should be visible at the beginning of the main() method.
- [x] CONSTANTS and their values should be visible outside the main() method.
- [x] Variable and its value should not be visible at the declaration line. It should be visible when we go to the next line (STEP_OVER).
- [x] Variable visibility inside control flows
    - [x] If/Else
    - [x] While
    - [x] Foreach
    - [x] Match
    Here we have to check variable count outside control flows. That is all variables should be visible when the debug navigator within the control flow.
- [x] Variable visibility test
    - [x] Enums
    - [x] Configurable variables
    
Fixes: https://github.com/ballerina-platform/ballerina-lang/issues/28730

### CODE REVIEW

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java [hunk 1]
```diff
@@ -171,7 +277,8 @@ public void localVariableVisibilityTest() {
         debugTestRunner.assertVariable(localVariables, "ĠĿŐΒȂɭ_ /:@[`{~⌤_json", "map<json>", "json");
     }
 
-    @Test
+    @Test(dependsOnMethods = "globalVariableVisibilityTest",
+        description = " Child variable visibility test for local variables at the last line of main() method")
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java:281]:
```suggestion
        description = "Child variable visibility test for local variables at the last line of main() method")
```

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java [hunk 2]
```diff
@@ -50,36 +52,140 @@ public void setup() throws BallerinaTestException {
...
-        // Asserts the number of globally visible variables (should be 11).
-        // Assert.assertEquals(globalVariables.size(), 11);
         // global constants
         // debugTestRunner.assertVariable(globalVariables, "nameWithoutType", "Ballerina", "string");
         // debugTestRunner.assertVariable(globalVariables, "nameWithType", "Ballerina", "string");
-
         debugTestRunner.assertVariable(globalVariables, "nameMap", "map", "map");
         debugTestRunner.assertVariable(globalVariables, "nilWithoutType", "()", "nil");
         debugTestRunner.assertVariable(globalVariables, "nilWithType", "()", "nil");
+        debugTestRunner.assertVariable(globalVariables, "RED", "()", "nil");
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java:173]: This assertion is wrong. The expected value in here should be "RED"

**Review by NipunaRanasinghe — APPROVED** [review]

File: tests/jballerina-debugger-integration-test/src/test/resources/project-based-tests/variable-tests/Config.toml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[variable_tests]
+    port = 9090
```
Comment (pramodya1994) [inline tests/jballerina-debugger-integration-test/src/test/resources/project-based-tests/variable-tests/Config.toml:2]: Do we need to keep a tab here?

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java (hunk 1 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java]: Fixed with [7a49014](https://github.com/ballerina-platform/ballerina-lang/pull/28809/commits/7a490140dba6ff648098b8ef1f3162245729dfed)

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java (hunk 2 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableVisibilityTest.java]: Fixed with [7a49014](https://github.com/ballerina-platform/ballerina-lang/pull/28809/commits/7a490140dba6ff648098b8ef1f3162245729dfed)

File: tests/jballerina-debugger-integration-test/src/test/resources/project-based-tests/variable-tests/Config.toml (hunk 3 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/test/resources/project-based-tests/variable-tests/Config.toml]: Fixed with [7a49014](https://github.com/ballerina-platform/ballerina-lang/pull/28809/commits/7a490140dba6ff648098b8ef1f3162245729dfed)

**Review by NipunaRanasinghe — APPROVED** [review]

**Review by pramodya1994 — APPROVED** [review]

**Pull request merged**